docs/diagrams/kubernetes_workflow.md

```diff
@@ -0,0 +1,1155 @@
+
+```mermaid
+%%{init: {'theme':'base', 'themeVariables': { 'actorBkg':'#000000', 'actorBorder':'#4a9eff', 'actorTextColor':'#ffffff', 'actorLineColor':'#4a9eff', 'signalColor':'#ffffff', 'signalTextColor':'#ffffff', 'labelBoxBkgColor':'#000000', 'labelBoxBorderColor':'#4a9eff', 'labelTextColor':'#ffffff', 'loopTextColor':'#ffffff', 'noteBkgColor':'#000000', 'noteTextColor':'#ffffff', 'noteBorderColor':'#ff6b35', 'activationBkgColor':'#1a1a1a', 'activationBorderColor':'#4a9eff', 'sequenceNumberColor':'#000000'}}}%%
+sequenceDiagram
+    participant KubeUnit
+    participant KubeAPI as Kubernetes API
+    participant Pod
+    participant LogStream as Log Stream
+    participant StdoutFile as stdout file
+    participant StdinFile as stdin file
+
+    Note over KubeUnit, StdinFile: Logger streaming method workflow
+
+    KubeUnit->>KubeUnit: RunWorkUsingLogger()
+
+    alt Resume existing pod
+        KubeUnit->>KubeAPI: Get pod by name
+        KubeAPI-->>KubeUnit: Existing pod
+        Note over KubeUnit: skipStdin = true
+    else Create new pod
+        KubeUnit->>KubeAPI: CreatePod()
+        KubeAPI-->>KubeUnit: Pod created
+        Note over KubeUnit: skipStdin = false
+    end
+
+    alt !skipStdin (new pod)
+        Note over KubeUnit: Wait for container Running state
+        
+        loop Check container state
+            KubeUnit->>KubeAPI: Get pod status
+            KubeAPI-->>KubeUnit: Container status
+            alt Container Waiting
+                KubeUnit->>KubeUnit: Sleep with backoff
+                KubeUnit->>KubeAPI: Retry get pod
+            else Container Terminated
+                KubeUnit->>KubeUnit: Mark as Failed
+            else Container Running
+                Note over KubeUnit: Break loop
+            end
+        end
+
+        KubeUnit->>KubeAPI: Create SPDY executor<br/>(SubResource attach)
+        KubeAPI-->>KubeUnit: Executor ready
+
+        par Stream stdin to pod
+            KubeUnit->>StdinFile: Open stdin file
+            StdinFile-->>KubeUnit: File reader
+            
+            KubeUnit->>KubeAPI: StreamWithContext(executor,<br/>StreamOptions{Stdin: reader})
+            
+            loop Stream stdin data
+                StdinFile->>KubeUnit: Read chunk
+                KubeUnit->>KubeAPI: Send chunk via SPDY
+                KubeAPI->>Pod: Forward to container stdin
+            end
+            
+            StdinFile->>KubeUnit: EOF
+            KubeUnit->>KubeAPI: Close stdin stream
+            KubeAPI->>Pod: Send EOF
+            Pod-->>KubeAPI: Stdin closed
+            KubeAPI-->>KubeUnit: Stream complete
+            
+            alt Stdin stream success
+                KubeUnit->>KubeUnit: UpdateBasicStatus(WorkStateRunning)
+            else Stdin stream error
+                KubeUnit->>KubeUnit: UpdateBasicStatus(WorkStateFailed)
+            end
+        and Stream stdout from pod
```

Comment:
Are `stdin` and `stdout` streaming at the same time? Hard to tell from the diagram if these happen sequentially or in parallel. I think they technically happen in parallel (concurrent goroutines) but I could see a reader interpreting we send stdin to completion first. \

Does `par` mean parallel?